includes/CLI/Plugin_Check_Command.php

```diff
@@ -289,6 +307,35 @@ public function list_checks( $args, $assoc_args ) {
 			)
 		);
 
+		// Filter by stability.
+		if ( ! empty( $options['stability'] ) ) {
+			$all_checks = array_filter(
+				$all_checks,
+				static function ( $item ) use ( $options ) {
+					return $options['stability'] === $item['stability'];
+				}
+			);
+		}
+
+		// Filter by categories.
+		if ( ! empty( $options['categories'] ) ) {
+			$all_checks = array_filter(
+				$all_checks,
+				static function ( $item ) use ( $options ) {
+					$categories = array_map( 'trim', explode( ',', $options['categories'] ) );
+					return array_intersect( $categories, $item['category'] );
+				}
+			);
+		}
+
+		// Display categories in better way.
+		array_walk(
+			$all_checks,
+			static function ( &$item ) {
+				$item['category'] = join( ', ', $item['category'] );
+			}
+		);
```

Comment:
Could you please explain why the changes needed?

Reply:
WP-CLI displays array value like this `["General", "Performance"]`. So we are converting to display like `general, performance`.

Reply:
I don't think this is needed. Before you had the `join()` call when creating the `$all_checks` list, and I think it can be kept that way.

The only thing you'll have to make sure of is that you do all the check filtering using the standard methods on the original check collection, _before_ creating the `$all_checks` variable for CLI output.